scripts/autotools/libmongocrypt/libmongocrypt.m4

```diff
@@ -0,0 +1,27 @@
+if test "$PHP_CLIENT_SIDE_ENCRYPTION" != "no"; then
+  AC_SUBST(MONGOCRYPT_ENABLE_TRACE, 1)
+  AC_SUBST(MONGOCRYPT_IS_POSIX, 1)
+  AC_SUBST(MONGOCRYPT_IS_WIN, 0)
+
+  PHP_MONGODB_LIBMONGOCRYPT_CFLAGS="$PHP_MONGODB_BUNDLED_CFLAGS -std=gnu99"
+
+  dnl Generated with: find src/libmongocrypt/src -maxdepth 1 -name '*.c' -print0 | cut -sz -d / -f 4- | sort -dz | tr '\000' ' '
+  PHP_MONGODB_MONGOCRYPT_SOURCES="mongocrypt-binary.c mongocrypt-buffer.c mongocrypt.c mongocrypt-cache.c mongocrypt-cache-collinfo.c mongocrypt-cache-key.c mongocrypt-ciphertext.c mongocrypt-crypto.c mongocrypt-ctx.c mongocrypt-ctx-datakey.c mongocrypt-ctx-decrypt.c mongocrypt-ctx-encrypt.c mongocrypt-key-broker.c mongocrypt-key.c mongocrypt-kms-ctx.c mongocrypt-log.c mongocrypt-marking.c mongocrypt-opts.c mongocrypt-status.c mongocrypt-traverse-util.c"
```

Comment:
Since this file is included before `CheckSSL.m4`, I think it's possible that we'd add these sources when `PHP_CLIENT_SIDE_ENCRYPTION` is "auto" and then fail to find a crypto library and disable FLE. I'll admit it's an edge case, but I'd prefer to inline this in `config.m4` after all of our includes and the existing libmongoc/libbson bundling code.

That would localize `PHP_MONGODB_ADD_SOURCES` and related macros in `config.m4` and be consistent with our existing m4 includes for libmongoc and libbson, which mainly do checks and call `AC_SUBST`. Doing so would also keep the "Generated with: find..." lines entirely in one file, too. This does mean we'd duplicate `$PHP_MONGODB_SSL` conditions in `config.m4` to add specific crypto sources, but I think it'd only be an if/elif for Darwin and OpenSSL/LibreSSL.

----

If you'd really like to keep this in a separate file, then I'd still suggest limiting libmongocrypt's `CheckSSL.m4` to definition code and moving its `ADD_SOURCES` calls to this file. Then, you could include _this_ m4 file towards the end of the the bundled condition in `config.m4` (after libbson and libmongoc sources are added). Note that doing so will likely require you to defer the `m4_popdef([_include])` call as well, since we'll need it for this.

Reply:
Great catch. I inlined the contents of libmongocrypt.m4 into config.m4, as I was planning to clean up the bundled build config in a later step anyways. We can then decide how we want to structure the includes.